### ISSUE TITLE
LibGit2Sharp update regressed some projects

### ISSUE THREAD

**Issue by AArnott** [AArnott/Nerdbank.GitVersioning#214]
After #210, NB.GV causes build failures for test projects. Some libraries seem to build OK though, oddly.

The repro below is for nerdbank.streams after updating to nb.gv 2.2.3.

```
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: System.Runtime.InteropServices.MarshalDirectiveException: StrictFilePathMarshaler must be used on a FilePath. [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.StrictFilePathMarshaler.MarshalManagedToNative(Object managedObj) [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.StubHelpers.MngdRefCustomMarshaler.ConvertContentsToNative(IntPtr pMarshalState, Object& pManagedHome, IntPtr pNativeHome) [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_repository_open(git_repository*& repository, FilePath path) [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.git_repository_open(String path) [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Repository..ctor(String path, RepositoryOptions options, RepositoryRequiredParameter requiredParameter) [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.GitExtensions.OpenGitRepo(String pathUnderGitRepo) [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/2.2.3/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner() [/home/andrew/git/nerdbank.streams/src/Nerdbank.Streams.Tests/Nerdbank.Streams.Tests.csproj]
```

**Comment by AArnott** [comment]
Closing in favor of #215
